lib/Dialect/FIRRTL/Transforms/LowerTypes.cpp

```diff
@@ -276,9 +276,9 @@ static MemOp cloneMemWithNewType(ImplicitLocOpBuilder *b, MemOp op,
       op.getRuw(), portNames, (op.getName() + field.suffix).str(),
       op.getNameKind(), op.getAnnotations().getValue(),
       op.getPortAnnotations().getValue(), op.getInnerSymAttr());
-  if (auto oldName = getInnerSymName(op))
-    newMem.setInnerSymAttr(hw::InnerSymAttr::get(StringAttr::get(
-        b->getContext(), oldName.getValue() + (op.getName() + field.suffix))));
+
+  assert(!op.getInnerSym() &&
+         "should already have produced error if sym present");
```

Comment:
Assert or error and exit?

Reply:
See arguments both ways, but since you asked let's make it an error.  It's unreachable presently so no test.